tomlkit/parser.py

```diff
@@ -466,27 +473,30 @@ def _parse_value(self):  # type: () -> Item
         """
         Attempts to parse a value at the current position.
         """
-        self.mark()
+        try:
+            return self._is_string(StringType.SLB)
```

Comment:
I am not sure about that. I would expect a `is_xxx()` method to return a boolean and not to execute an action.

I think the previous way of doing is much clearer.

```python
if c == StringStype.SLB.value:
    self._parse_basic_string()
```

That's what I did here https://github.com/sdispater/tomlkit/commit/428a9b2c4f785bd17e81b2940383a3dc7bd6bed2#diff-58b7a35cabb00a194ef1dc33829a5975R473

You will see that I also removed the need of raising a `Restore` exception since it makes the code more complex than it should be.